File: HandsOn-ClockDigital-Java/app/src/main/java/com/example/clockdigital/MainActivity.java
package com.example.clockdigital;

import android.os.Bundle;

import android.view.View;
import android.widget.TextView;
import android.widget.TextClock;
import android.widget.Button;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private TextView txtExibirHoraUpdate;
    private TextClock txtHoraAtual, txtHora24PM;
    private Button btnAtualizarHoraAtual;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        inicializarComponentes();
        capturarHora();
    }

    private void capturarHora() {
        btnAtualizarHoraAtual.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                txtExibirHoraUpdate.setText("Hora: "+txtHora24PM.getText());
            }
        });
    }

    private void inicializarComponentes() {
        txtExibirHoraUpdate = findViewById(R.id.txtExibirHoraUpdate);
        txtHoraAtual = findViewById(R.id.txtHoraAtual);
        txtHora24PM = findViewById(R.id.txtHora24PM);
        btnAtualizarHoraAtual = findViewById(R.id.btnAtualizarHora);
    }
}

File: HandsOn-ClockDigital-Kotlin/app/src/main/java/com/example/handson_clockdigital/MainActivity.kt
package com.example.handson_clockdigital

import android.view.View
import android.widget.TextView
import android.widget.TextClock
import android.widget.Button
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private var txtHoraAtual : TextClock? = null;
    private var txtHora24pm : TextClock? = null;
    private var txtHora : TextView? = null;
    private var btnUpdateHora : Button? = null;


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        iniciarComponentes();
    }

    private fun iniciarComponentes() {
        txtHora = findViewById<TextView>(R.id.txtExibirHoraUpdate);
        txtHoraAtual = findViewById(R.id.txtHoraAtual);
        txtHora24pm = findViewById(R.id.txtHora24PM);
        btnUpdateHora = findViewById(R.id.btnAtualizarHora);
    }

    fun atualizarHoraClick(view: View) {
        txtHora!!.text = ("Hora: " + txtHora24pm!!.text);
    }
}